### PR TITLE
Table: Fix row count during filter

### LINKED ISSUE #71568: row count is wrong
### What happened?

when we apply a filter inside the panel, the count of rows continues counting all items outside the filter

### What did you expect to happen?

when applying a filter within the panel, the Rows should only show the count of the amount with the filter

### Did this work before?

no

### How do we reproduce it?

1.apply filter inside the panel
2. the rows must count the quantity of the items again
3.rows display the correct values

### Is the bug inside a dashboard panel?

all panels

### Environment (with versions)?

Grafana:9.5
OS:Enterprise 
Browser: chrome

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by drew08t**
During column filter in table, row count is not updating properly.

- [x] Row count in footer
- [x] Pagination

Before:

https://github.com/grafana/grafana/assets/60050885/763fc8ce-2f98-4b79-a92a-b4a0f49928fb

After:

https://github.com/grafana/grafana/assets/60050885/4b4311b5-a779-4788-a77a-1a457c2e60c0

Fixes #71568

### CODE REVIEW

**Review by nmarrs — APPROVED** [review]
testing locally everything seems to work well 💯 - but I think it wouldn't hurt to have @codeincarnate or @baldm0mma take a quick look to make sure this change makes sense 👍

File: packages/grafana-ui/src/components/Table/Table.tsx [hunk 1]
```diff
@@ -215,7 +215,7 @@ export const Table = memo((props: Props) => {
 
     if (isCountRowsSet) {
       const footerItemsCountRows: FooterItem[] = [];
-      footerItemsCountRows[0] = headerGroups[0]?.headers[0]?.filteredRows.length.toString() ?? data.length.toString();
+      footerItemsCountRows[0] = rows.length.toString() ?? data.length.toString();
```
[the remark this answers is not in the document]
Comment (drew08t) [reply packages/grafana-ui/src/components/Table/Table.tsx:218]: Same thought!

[the remark this answers is not in the document]
Comment (codeincarnate) [reply packages/grafana-ui/src/components/Table/Table.tsx:218]: I can't comment on original usage but I don't _think_ so. `headerGroups` is from react table state so in the old code it's examining it's internal state AFAIK. This seems fine to me :smile:

**Review by codeincarnate — APPROVED** [review]
Looks go to me! As far as I know there isn't a problem with the code as is

**Pull request merged**